File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.cpp
#include <string>
#include <vector>
#include <queue>
using namespace std;

vector<int> solution(int k, vector<int> score) {
    vector<int> answer;
    priority_queue<int, vector<int>, greater<int>> que;
    
    for(int i =0; i<score.size();i++)
    {
        if(que.size() == k)
        {
            if(que.top() <score[i])
            {
                que.pop();
                que.push(score[i]);
            }
        }
        else que.push(score[i]);
        
        answer.push_back(que.top());
    }
    return answer;
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.cs
using System;
using System.Linq;
using System.Collections.Generic;
public class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.Length];
        List<int> res = new List<int>();
        for(int i = 0; i<score.Length; i++)
        {
            res.Add(score[i]);
            res.Sort();
            if (res.Count > k)
            {
                res.RemoveAt(0);
            }
            answer[i] = res[0];
        }
        return answer.ToArray();
    }
}